lib/util/npm-util.js

```diff
@@ -34,6 +36,8 @@ function findPackageJson(startDir) {
         if (!fs.existsSync(pkgFile) || !fs.statSync(pkgFile).isFile()) {
             dir = path.join(dir, "..");
             continue;
+        } else if (fs.existsSync(path.join(dir, "yarn.lock"))) {
```

Comment:
To make `npm` unavailable with node installation, the developer must build Node from source or somehow else hack the installation. I think such people are not target user of `eslint --init`, and implementing actual fallback from `npm` to `yarn` will complicate the whole codebase for a really rare use case.